.github/workflows/demos-docker-build.yaml

```diff
@@ -37,12 +37,12 @@ jobs:
         python-version: [3.7]
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v3
       with:
         python-version: ${{ matrix.python-version }}
-    - uses: actions/cache@v2
+    - uses: actions/cache@v3
```

Comment:
we could potentially remove this by using https://github.com/actions/setup-python#caching-packages-dependencies above